src/language-js/comments.js

```diff
@@ -954,8 +955,7 @@ function willPrintOwnComments(path /*, options */) {
       (isJsxNode(node) ||
         hasFlowShorthandAnnotationComment(node) ||
         (isCallExpression(parent) &&
-          (hasFlowAnnotationComment(node.leadingComments) ||
-            hasFlowAnnotationComment(node.trailingComments))))) ||
+          hasFlowAnnotationComment(getComments(node))))) ||
```

Comment:
I saw this `hasFlowAnnotationComment` function only checks the first comment, what if the "FlowAnnotationComment" is attached as trailing comment, and there are non-related leading comments, will it work as expected?

I don't know syntax about flow.

Reply:
I don't know syntax about flow too...

The below code works fine?

```suggestion
          hasFlowAnnotationComment(getComments(node, CommentCheckFrag.Leading) || hasFlowAnnotationComment(getComments(node, CommentCheckFrag.Trailing))))) ||
```

Reply:
See https://github.com/prettier/prettier-regression-testing/issues/68#issuecomment-887241644. At least source code of Babel has no problem. (But #11247 causes ugly formatting...)

Reply:
> The below code works fine?

I guess.

Reply:
bd680b9

Reply:
Good job.